File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/ThreadPool.java
package com.ninpou.packetcapture.core.util.common;

import java.util.concurrent.Executor;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

public class ThreadPool {
    private final Executor executor;

    private ThreadPool() {
        executor = new ThreadPoolExecutor(1, 4,
                10L, TimeUnit.MILLISECONDS,
                new LinkedBlockingQueue<Runnable>(1024));
    }

    public static ThreadPool getInstance() {
        return Singleton.instance;
    }

    public void execute(Runnable run) {
        executor.execute(run);
    }

    static class Singleton {
        static ThreadPool instance = new ThreadPool();
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/ThreadPool.kt
package com.ninpou.packetcapture.core.util.common

import java.util.concurrent.Executor
import java.util.concurrent.LinkedBlockingQueue
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

class ThreadPool private constructor() {
    private val executor: Executor
    fun execute(run: Runnable?) {
        executor.execute(run)
    }

    internal object Singleton {
        var instance = ThreadPool()
    }

    companion object {
        @JvmStatic
        val instance: ThreadPool
            get() = Singleton.instance
    }

    init {
        executor = ThreadPoolExecutor(1, 4,
                10L, TimeUnit.MILLISECONDS,
                LinkedBlockingQueue(1024))
    }
}